vowpalwabbit/example.h

```diff
@@ -158,11 +158,35 @@ struct example : public example_predict  // core example datatype.
   //     them. " "Standalone examples are by definition always in use.")
   bool in_use = true;
 
-  // This object is only used because atomic signaling is not available in C++11
-  example_lock ex_lock;
+  // // This object is only used because atomic signaling is not available in C++11
+  // example_lock ex_lock;
 };
 VW_WARNING_STATE_POP
 
+struct example_vector  // core example datatype.
```

Comment:
With the current changes, if I were to use the second method, I should just move the `example_lock` object from the `example_vector`, back to the `example` struct. right?

Reply:
the v_array usage in the parser is mostly a legacy thing. The multi_ex definition used to be `v_array<example*>` but it was converted to `vector<example*>` sometime in the last year or so. However, since the parser and learner's usage of them are completely separate, we never went and refactored the parser.